server/src/main/java/org/apache/druid/metadata/MetadataSegmentManager.java

```diff
@@ -98,6 +98,15 @@
 
   Collection<String> getAllDataSourceNames();
 
+  /**
+   * Returns a collection of overshadowed segments
+   *
+   * Will return null if we do not have a valid snapshot of segments yet (perhaps the underlying metadata store has
+   * not yet been polled.)
+   */
+  @Nullable
+  Collection<DataSegment> findOvershadowedSegments();
```

Comment:
A method which is a simple getter shouldn't be called `find...`, implying high cost.